harper-ls/src/backend.rs

```diff
@@ -407,6 +435,34 @@ impl LanguageServer for Backend {
 
     async fn did_close(&self, _params: DidCloseTextDocumentParams) {}
 
+    async fn did_delete_files(&self, params: DeleteFilesParams) {
```

Comment:
It looks like you're just listening for files that have been deleted by the editor. If a file is deleted separately (i.e. via File Explorer or bash), this won't pick it up. 

Would you mind adding listeners to watch for other kinds of deletions? I'd be happy to make it happen, just won't be able to for a bit.

Reply:
Yeah, this only works if a file is deleted within the editor. I'll look for a way so it'd also pick up deletions that happen outside of the editor. I'll turn this PR into a draft until it's ready.